owncloudApp/src/main/java/com/owncloud/android/ui/activity/FileDisplayActivity.kt

```diff
@@ -1176,55 +1176,67 @@ class FileDisplayActivity : FileActivity(),
-                    SynchronizeFileUseCase.SyncType.AlreadySynchronized -> showSnackMessage(getString(R.string.sync_file_nothing_to_do_msg))
+                    SynchronizeFileUseCase.SyncType.AlreadySynchronized -> {
+                        if (fileWaitingToPreview != null) {
+                            startPreview(fileWaitingToPreview)
+                            fileWaitingToPreview = null
+                        } else {
+                            showSnackMessage(getString(R.string.sync_file_nothing_to_do_msg))
+                        }
+                    }
                     is SynchronizeFileUseCase.SyncType.ConflictDetected -> {
                         val showConflictActivityIntent = Intent(this, ConflictsResolveActivity::class.java)
                         showConflictActivityIntent.putExtra(ConflictsResolveActivity.EXTRA_FILE, file)
                         startActivity(showConflictActivityIntent)
                     }
-                    is SynchronizeFileUseCase.SyncType.DownloadEnqueued -> showSnackMessage("Download enqueued")
+                    is SynchronizeFileUseCase.SyncType.DownloadEnqueued -> {
+                        if (fileWaitingToPreview != null) {
+                            showSnackMessage(getString(R.string.new_remote_version_found_msg))
+                            startSyncThenOpen(fileWaitingToPreview!!)
+                            fileWaitingToPreview = null
+                        } else {
+                            showSnackMessage(getString(R.string.download_enqueued_msg))
+                        }
+                    }
                     SynchronizeFileUseCase.SyncType.FileNotFound -> {
                         /** Nothing to do atm. If we are in details view, go back to file list */
                     }
-                    is SynchronizeFileUseCase.SyncType.UploadEnqueued -> showSnackMessage("Upload enqueued")
+                    is SynchronizeFileUseCase.SyncType.UploadEnqueued -> showSnackMessage(getString(R.string.upload_enqueued_msg))
                     null -> TODO()
                 }
             }
             is UIResult.Error -> {
-                when (uiResult.error) {
-                    is UnauthorizedException -> {
-                        launch(Dispatchers.IO) {
-                            val credentials = AccountUtils.getCredentialsForAccount(MainApp.appContext, account)
-
-                            launch(Dispatchers.Main) {
-                                if (credentials is OwnCloudBearerCredentials) { // OAuth
-                                    showRequestRegainAccess()
-                                } else {
-                                    showRequestAccountChangeNotice(getString(R.string.auth_failure_snackbar), false)
+                if (fileWaitingToPreview != null) {
+                    startPreview(fileWaitingToPreview)
+                    fileWaitingToPreview = null
+                } else {
```

Comment:
If we get an error when syncing because credentials are not valid anymore, we probably want to ask to the user about login again. So I don't think we should add that into an else

Reply:
👍